scripts/split-by-solvency.ts

```diff
@@ -0,0 +1,71 @@
+import { SuperTokenList, TokenInfo } from "..";
+import tokenList from "../superfluid.tokenlist.json";
+import fs from "fs";
+import { validate, validateUnderlyingTokens } from "../utils";
+
+const createFilteredList = async (tokenList: SuperTokenList, tier: string) => {
+  const underlyingTokens = tokenList.tokens.filter((token) =>
+    token.tags?.includes("underlying")
+  );
+
+  const filteredByTier = tokenList.tokens.filter((token) =>
+    token.tags?.includes(tier)
+  );
+
+  const tokens: TokenInfo[] = [];
+
+  filteredByTier.forEach((token) => {
+    tokens.push(token);
+
+    if (
+      ["Wrapper", "Native Asset"].includes(
+        token.extensions?.superTokenInfo?.type ?? ""
+      )
```

Comment:
Approaching from `token.extensions?.superTokenInfo?.underlyingTokenAddress` is slightly easier in the sense that you don't even have to know the type of the token. If it has an underlying token then include it in the list.

Reply:
that's not a common property between the types.